sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -450,7 +455,11 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
   }
 
   override def tableExists(db: String, table: String): Boolean = withClient {
-    client.getTableOption(db, table).isDefined
+    try {
+      client.getTableOption(db, table).isDefined
+    } catch {
+      case e: AnalysisException if e.message.contains("Hive index table is not supported") => true
```

Comment:
this looks hacky. Actually why we use `getTableOption` to implement the semantic of `tableExists`? I'd like to make `HiveClient` provide the `tableExists` API directly.